pkg/kube/inject/inject.go

```diff
@@ -971,6 +971,9 @@ func cleanMeshConfig(v proto.Message) proto.Message {
 	if cpy.SdsUdsPath == defaults.SdsUdsPath {
 		cpy.SdsUdsPath = ""
 	}
+	if cpy.IngressControllerMode == defaults.IngressControllerMode {
+		cpy.IngressControllerMode = meshconfig.MeshConfig_UNSPECIFIED
```

Comment:
What does this mean?

Reply:
Will this change `MeshConfig_STRICT` to UNSPECIFIED? And makes it invalid matching the default case?

```
	switch mesh.IngressControllerMode {
	case meshconfig.MeshConfig_OFF:
		return false
	case meshconfig.MeshConfig_STRICT:
		return exists && class == mesh.IngressClass
	case meshconfig.MeshConfig_DEFAULT:
		return !exists || class == mesh.IngressClass
	default:
		log.Warnf("invalid ingress synchronization mode: %v", mesh.IngressControllerMode)
		return false
	}
```

Reply:
no, because the default is no unspecified, it's STRICT. See the associated test, if will be unmarshalled as strict

Reply:
@howardjohn I donot think we need to cleanup `IngressControllerMode` for injector, i can hardly think why sidecar need this flag

Reply:
It doesn't but we pass it the whole mesh config

Reply:
this isn't about setting the value, it's about making this removed from the json output so that we only include it if it's not the default setting. This reduced the pod spec size

Reply:
IC